firmware/cmake/embedded.cmake

```diff
@@ -180,6 +180,11 @@ function(embedded_binary
     add_custom_target(${HEX_FILE} ALL DEPENDS ${HEX_PATH})
     add_custom_target(${ASM_FILE} ALL DEPENDS ${ASM_PATH})
 
+    add_custom_command(TARGET ${ELF_NAME} POST_BUILD
+    COMMAND ${PYTHON_COMMAND} ${SCRIPTS_DIR}/clang_format/fix_formatting.py
+    WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}
+    COMMENT "Formatting Source Files")
+    
```

Comment:
formatting after building is weird, why are we doing this? 
what's wrong with running script manually: 
`python3 scripts/clang-format/fix_formatting.py`

Reply:
@philam2001 wants it :)

Reply:
why run manually when you can format + build ur files at the same time ;D but hm I'm pretty sure it formats before actually building, if anything it just has the built target before

Reply:
Could also do it when loading the cmake, it was there originally. But yea like it'd be nice to be able to have it auto clang format as you are loading ur cmake and building to remove redundant steps to format.